src/controllers/cart.ts

```diff
@@ -1,23 +1,20 @@
 import { Request, Response } from "express";
-import {
-  createUserCart,
-  deleteUserCart,
-  getUserCart,
-  updateUserCart,
-} from "../services/cart";
-import { Cart, User } from "../types/types";
+import { CartItem, User } from "../types/types";
 import { updateCartSchema } from "../helpers/validations";
 import {
   RESPONSE_CODE_BAD_REQUEST,
   RESPONSE_CODE_OK,
   RESPONSE_CODE_SERVER_ERROR,
 } from "../constants/responseCodes";
+import { Cart } from "../entities/Cart";
+import { createUserCart, getUserCart } from "../services/cart";
 
 export async function createCart(req: Request, res: Response) {
   try {
     const userId = (req as any).user.id;
```

Comment:
(req as any) - this loosing all advantages of typescript here, so all next manipulations like:
.find, .findOne, ... etc 
are return 'any'.